### PR TITLE
migrate channel header dropdown comp to functional comp

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

Migrate the channel header dropdown component from class based to function based component.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24752

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by jespino — APPROVED** [review]
Nice! Thanks @tanmaythole! :confetti_ball:

File: webapp/channels/src/components/channel_header_dropdown/channel_header_dropdown.tsx [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-            <Menu
-                id='channelHeaderDropdownMenu'
-                ariaLabel={localizeMessage('channel_header.menuAriaLabel', 'Channel Menu').toLowerCase()}
-            >
-                <ChannelHeaderDropdownItems isMobile={false}/>
-            </Menu>
-        );
-    }
-}
+const ChannelHeaderDropdown: FC = () => (
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/channel_header_dropdown.tsx:12]: There is no need for the explicit typing.

**Review by larkox — APPROVED** [review]
Great work. Thanks!

**Comment by larkox** [comment]
There seems to be some snapshot failings. Probably just updating the snapshots should fix all the issues: https://github.com/mattermost/mattermost/actions/runs/6439399261/job/17488080410?pr=24793

**Comment by tanmaythole** [comment]
I don't know why testcase fails, it work perfectly in local machine

**Comment by jespino** [comment]
@tanmaythole I don't think is related to your changes, let see if merging the `master` branch here, we get everything passing.

**Comment by ctlaltdieliet** [comment]
I tested it on test-server and it is working!
Screenshot from 2023-10-12 16-01-01

**Review by ctlaltdieliet — COMMENTED** [review]
lgtm

**Review by ctlaltdieliet — APPROVED** [review]
lgtm
It works on my computer :smile:

**Comment by jespino** [comment]
Thanks @ctlaltdieliet! You are everywhere! :sparkling_heart:

**Comment by jespino** [comment]
@lindy65 looks like there are 2 test failing, but looks like they are not related to this changes.

**Comment by lindy65** [comment]
Thanks @jespino 👍

**Review by lindy65 — APPROVED** [review]
Thanks @tanmaythole - Looks good to me 👍 

Thanks @ctlaltdieliet for QAing!

cc @larkox

**Comment by larkox** [comment]
Thanks for the contribution :tada:

**Pull request merged**